hexrd/ui/image_file_manager.py

```diff
@@ -143,49 +144,17 @@ def is_hdf(self, extension):
 
         return False
 
-    def guess_path(self, f):
-        h5 = h5py.File(f, 'r')
-        location = ''
-        dataname='DATA'
-        if 'ATTRIBUTES' in h5:
-            if ('PINHOLE_ORIENTED_IMAGE' in h5['ATTRIBUTES']) and ('TARGET_ORIENTED_IMAGE' in h5['ATTRIBUTES']):
-                h5.close()
-                try:
-                    imageseries.open(f, 'hdf5', path='ATTRIBUTES/TARGET_ORIENTED_IMAGE/DATA', dataname=dataname)
-                except:
-                    imageseries.open(f, 'hdf5', path='ATTRIBUTES/PINHOLE_ORIENTED_IMAGE/DATA', dataname=dataname)
-            elif 'PSL_FADE_CORR_IMAGE' in h5['ATTRIBUTES']:
-                print(f'PSL_FADE_CORR_IMAGE in ATTRIBUTES')
-                location = 'PSL_FADE_CORR_IMAGE'
-            elif 'HDR_IMAGE' in h5['ATTRIBUTES']:
-                location = 'HDR_IMAGE'
-            elif 'CORR_IMAGE' in h5['ATTRIBUTES']:
-                location = 'CORR_IMAGE'
-            elif 'TIME_ADJUSTED_IMAGE' in h5['ATTRIBUTES']:
-                location = 'TIME_ADJUSTED_IMAGE'
-            elif 'PSL_IMAGE' in h5['ATTRIBUTES']:
-                location = 'PSL_IMAGE'
-        elif 'DATA' in h5:
-            location = 'DATA'
-        h5.close()
-        path=f'ATTRIBUTES/{location}/DATA'
-        imageseries.open(f, 'hdf5', path=path, dataname=dataname)
-        return path, dataname
-
     def path_exists(self, f):
-        try:
-            path, dataname = HexrdConfig().hdf5_path
-            imageseries.open(f, 'hdf5', path=path, dataname=dataname)
-            self.path = HexrdConfig().hdf5_path
-            return True
-        except:
-            try:
-                path, dataname = self.guess_path(f)
+        all_paths = [*constants.KNOWN_HDF5_PATHS, HexrdConfig().hdf5_path]
+        h5 = h5py.File(f, 'r')
```

Comment:
Let's use a context manager.

```python
with h5py.File(f, 'r') as hf:
    ...
```

Then indent all of the below code and remove `h5.close()` in each case.

The context manager is better here because if an exception is raised before the file is closed, it may never be closed, and it may make the user's HEXRDGUI session not work anymore.

Reply:
Fixed